### ISSUE TITLE
Multi-line comment indentation + compact output makes code generation very slow

### ISSUE THREAD

**Issue by jbt** [babel/babel#2008]
If I have a really big file like:

```js
var foo = class {
	doesntReallyMatter(){}
}
... few thousand lines of things ...

/*
 @license
 */
... maybe some other stuff
```

and I compile it with `compact:true` (or the auto-compact-erizer kicks in because it's >100k, with or without `comments:false` because `@license` is always preserved), code generation takes ages (many minutes, versus seconds if the license comment isn't there)

It looks like it's coming from the [trim-right in `Buffer#get`](https://github.com/babel/babel/blob/1f851153fd0d09f13d17a6d7ecf2e1404e418c30/packages/babel/src/generation/buffer.js#L24) (replacing that line with `return this.buf` makes everything fast again).

By the look of it, trim-right uses [a regex that's quadratically bad](https://github.com/sindresorhus/trim-right/blob/master/index.js#L3) if you have massive expanses of whitespace that _aren't_ at the end of the string, which is what happens in this case because the output looks something like this (with potentially enormous amounts of whitespace to align the comment):

```js
foo something [lots and lots of stuff, many hundreds of thousands of characters] /*
                                                                                  @license
                                                                                  */ rest of the things
```

I'm not sure what you'd consider to be the most elegant solution here - changing `Buffer#get` to something like:

```js
get(){
	var tail = this.buf.length;
	while(/[\s\uFEFF\xA0]/.test(this.buf.charAt(tail - 1))){
		tail--;
	}
	return this.buf.slice(0, tail);
}
```

... seems to work for me (although it may or may not have an off-by-one error because I've written it fairly quickly). Alternatively getting rid of the indentation of comments in compact mode would probably achieve the same effect.

**Comment by sebmck** [comment]
Thanks for the awesome report @jbt. It's worth doing both of your suggestions, I've submitted a PR to `trim-right` with your suggested modification and will make the necessary changes to not align comment columns.

**Comment by sebmck** [comment]
Just released 5.7.1 which defaults `adjustMultilineComment` to `false` when `compact` is `true`. Once/if my `trim-right` patch is merged you should see a perf boost to the code generator too. Thanks! :blush:

**Comment by jbt** [comment]
Hurray many speeds, excellent!
